src/main/java/org/broadinstitute/hellbender/utils/Utils.java

```diff
@@ -871,4 +866,42 @@ public static void warnOnNonIlluminaReadGroups(final SAMFileHeader readsHeader,
             logger.warn("This tool has only been well tested on ILLUMINA-based sequencing data. For other data use at your own risk.");
         }
     }
+
+    /**
+     * Returns true iff we can make instances of this class.
+     */
+    public static boolean canMakeInstances(final Class<?> clazz) {
+        return clazz != null &&
+                !clazz.isPrimitive()  &&
+                !clazz.isSynthetic()  &&
+                !clazz.isInterface()  &&
+                !clazz.isLocalClass() &&
+                !Modifier.isAbstract(clazz.getModifiers());
+    }
+
+    /**
+     * Finds and creates objects of all concrete subclasses of the given class in the package.
+     * The public no-arg constructor is called to create the objects.
```

Comment:
done. I removed `AutoDiscoverable` - we may want to instantiate classes that we don't own.